### PR TITLE
Fix DebugBar session issue, add DebugBar\JoomlaHttpDriver

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue #37784 .

### Summary of Changes

DebugBar require $_SESSION to work. And because API Application uses "Runtime session" it crashes.
The solution is to use the same Session interface that Joomla Application. For this I have added JoomlaHttpDriver for DebugBar.

Bonus: Now debug plugin also collect information for API requests under cache/plg_system_debug_api. 
However currently in not possible to view in DebugBar, but that for another PR.

### Testing Instructions
Please follow #37784 

### Actual result BEFORE applying this Pull Request
An error

### Expected result AFTER applying this Pull Request
No error

### Documentation Changes Required
none

### CODE REVIEW

File: plugins/system/debug/debug.php [hunk 1]
```diff
@@ -95,7 +97,7 @@ class PlgSystemDebug extends CMSPlugin implements SubscriberInterface
 	/**
 	 * Application object.
 	 *
-	 * @var    CMSApplicationInterface
+	 * @var    CMSApplication|ConsoleApplication
```
Comment (laoneo) [inline plugins/system/debug/debug.php]: Why?

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 81914bc40a1cf9c3039a7780af4a203f36e05c05<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37797">issues.joomla.org/tracker/joomla-cms/37797</a>.</sub>

File: plugins/system/debug/debug.php (hunk 1 above)
Comment (Fedik) [reply plugins/system/debug/debug.php]: yea, make no sense, I have changed in diferent way now

**Comment by Fedik** [comment]
I have corected AppInterfaces, please re-test

**Review by laoneo — APPROVED** [review]

**Comment by alikon** [comment]
ok retested

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 2b77344063a154ac669ed5e5119f40e6279be791<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37797">issues.joomla.org/tracker/joomla-cms/37797</a>.</sub>

**Review by wilsonge — APPROVED** [review]

File: plugins/system/debug/src/JoomlaHttpDriver.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	/**
+	 * Checks if the session is started
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function isSessionStarted()
+	{
+		return $this->app->getSession()->isStarted();
```
Comment (wilsonge) [inline plugins/system/debug/src/JoomlaHttpDriver.php]: do we not also need to check for `SessionAwareWebApplicationInterface` to do this (and in the other methods). Given we're doing the WebApplicationInterface check in the above method

Comment (Fedik) [reply plugins/system/debug/src/JoomlaHttpDriver.php]: CMSApplicationInterface implements it already, so I skiped

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 2b77344063a154ac669ed5e5119f40e6279be791<br /><br />Tested successfully in Joomla 4.1.4-rc2-dev of 22 May in Wampserver 3.2.8 using PHP 8.0.15 and Postman.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37797">issues.joomla.org/tracker/joomla-cms/37797</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37797">issues.joomla.org/tracker/joomla-cms/37797</a>.</sub>

File: plugins/system/debug/src/JoomlaHttpDriver.php (hunk 2 above)
Comment (wilsonge) [reply plugins/system/debug/src/JoomlaHttpDriver.php]: No it doesn't only CMSWebApplicationInterface https://github.com/joomla/joomla-cms/blob/4.1-dev/libraries/src/Application/CMSWebApplicationInterface.php#L23

**Review by wilsonge — DISMISSED** [review]
Blocking merge based on session discussion for now

File: plugins/system/debug/src/JoomlaHttpDriver.php (hunk 2 above)
Comment (Fedik) [reply plugins/system/debug/src/JoomlaHttpDriver.php]: Yes, but it already extends it, 
I will change

**Comment by Fedik** [comment]
I have updated the code, to check app session

File: plugins/system/debug/src/JoomlaHttpDriver.php [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	public function getSessionValue($name)
+	{
+		if ($this->session)
+		{
+			return $this->session->get($name);
+		}
+		else
+		{
+			return $this->dummySession[$name] ?? null;
+		}
```
Comment (Quy) [inline plugins/system/debug/src/JoomlaHttpDriver.php]:
```suggestion

		return $this->dummySession[$name] ?? null;
```

**Comment by richard67** [comment]
@alikon @toivo Could you test again? Thanks in advance.

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 21cfb5a7e65d504c6fb5d61d3683b0434668adf4<br /><br />Tested successfully in Joomla 4.1.4 in Wampserver 3.2.8 using PHP 8.0.15.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37797">issues.joomla.org/tracker/joomla-cms/37797</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 21cfb5a7e65d504c6fb5d61d3683b0434668adf4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37797">issues.joomla.org/tracker/joomla-cms/37797</a>.</sub>

**Comment by richard67** [comment]
Thanks for testing. RTC still set.

@wilsonge Could you approve the changes so GitHub doesn’t show an unresolved review anymore?

**Comment by bembelimen** [comment]
Thx

**Pull request merged**